addons/project_mail_plugin/controllers/mail_plugin.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+
+from odoo.http import request
+
+from odoo.addons.mail_plugin.controllers import mail_plugin
+
+_logger = logging.getLogger(__name__)
+
+
+class MailPluginController(mail_plugin.MailPluginController):
+
+    def _fetch_partner_tasks(self, partner, limit=5, offset=0):
+        """
+        Returns an array containing partner tasks, each task will have the following structure :
+        {
+            id: the task's id,
+            name: the task's name,
+            project_name: the task's project name
+            company_id: the id of the company related to the record,
+        }
+        """
+
+        partner_tasks = request.env['project.task'].search(
+            [('partner_id', '=', partner.id)], offset=offset, limit=limit)
+
+        return [
+            {
+                "task_id": task.id,
+                "name": task.name,
+                "project_name": task.project_id.name,
+                "company_id": task.company_id.id,
+            }
+            for task in partner_tasks
+        ]
+
+    def _get_contact_data(self, partner):
+        contact_values = super(MailPluginController, self)._get_contact_data(partner)
+        if not partner:
+            contact_values['tasks'] = []
+        else:
+            contact_values['tasks'] = self._fetch_partner_tasks(partner)
```

Comment:
Is there any reason not to inline the code from "_fetch_partner_tasks" here?
Is it meant to be overridden somewhere?

I don't think the docstring of the method is a valid reason since it's pretty straight forward, you can see the structure by just looking at the constructed dict.

We usually don't split the code into methods unless there is a good reason to do so (add modularity by making it overridable / initial method becomes very lengthy / allows to explain with elaborated docstring / ...).
Indeed, this tends to reduce readability (you have to grep the method to keep reading the code).

Reply:
Yes, I think that we can get rid of that method, since it is not used elsewhere.